src/openrct2/ride/TrackDesign.cpp

```diff
@@ -528,7 +528,9 @@ rct_string_id TrackDesign::CreateTrackDesignScenery()
             }
         }
 
-        CoordsXY sceneryMapPos{ scenery.x * 32 - gTrackPreviewOrigin.x, scenery.y * 32 - gTrackPreviewOrigin.y };
+        // Cast the value into a uint8_t as this value is not signed yet.
+        CoordsXY sceneryMapPos{ ((uint8_t)scenery.x) * 32 - gTrackPreviewOrigin.x,
```

Comment:
Use `static_cast`

Reply:
Also use COORDS_XY_STEP instead of 32. (Bit of camp floor cleaning ;))